libs/fluid-elements/button-group/src/lib/button-group-item/button-group-item.ts

```diff
@@ -0,0 +1,393 @@
+/**
+ * @license
+ * Copyright 2020 Dynatrace LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  html,
+  css,
+  LitElement,
+  CSSResult,
+  TemplateResult,
+  property,
+  customElement,
+  unsafeCSS,
+} from 'lit-element';
+import { classMap } from 'lit-html/directives/class-map';
+import { FluidButtonGroupItemCheckedChangeEvent } from '../button-group-events';
+
+import { ENTER, SPACE, TAB } from '@dynatrace/shared/keycodes';
+
+import {
+  fluidDtText,
+  FLUID_SPACING_2X_SMALL,
+  FLUID_SPACING_SMALL,
+} from '@dynatrace/fluid-design-tokens';
+
+let uniqueCounter = 0;
+
+/**
+ * A basic representation of the button-group-item
+ * @element fluid-button-group-item
+ * @slot - Default slot lets the user provide a label for the button-group-item.
+ * @fires checkedChange - Event that is being fired when the button-group-item checked state changes due
+ * to user interaction.
+ * @cssprop --fluid-button-group-item--label-color - Customize the label color.
+ * @cssprop --fluid-button-group-item--radio-hover-color - Customize the color of the button-group-item nob when hovering.
+ * @cssprop --fluid-button-group-item--radio-active-color - Customize the color of the button-group-item nob when active.
+ */
+@customElement('fluid-button-group-item')
+export class FluidButtonGroupItem extends LitElement {
+  /**
+   * Unique identifier used for the id and label connection
+   * within the checkbox.
+   */
+  private _unique = `fluid-checkbox-${uniqueCounter++}`;
+
+  /** Reference to the input container */
+  private _inputElement: HTMLInputElement;
+
+  constructor() {
+    super();
+  }
+
+  /** Styles for the button-group component */
+  static get styles(): CSSResult {
+    return css`
+      :host {
+        /**
+        * Legibility definitions should probably be
+        * shipped or imported from a core
+        */
+        -webkit-font-smoothing: antialiased;
+        -moz-osx-font-smoothing: grayscale;
+        text-rendering: optimizeLegibility;
```

Comment:
why do we specify this on a per component level?

Reply:
Because at this point there is no core yet (if there ever will be one for css). It's more likely that this will be provided by some sort of wrapper custom-element (similar to the design system provider from fast https://fast.design/docs/components/design-system-provider)